backend/src/routes/payments.js

```diff
@@ -32,6 +32,33 @@ const addBalance = async (req, res) => {
     }
 };
 
+const getPaymentDetails = async (req, res) => {
+    try {
+        const ordernumber = req.query.ordernumber;
+        const user = req.user;
+        if (ordernumber === null) {
+            return res.status(422).json('Order number is not valid');
+        }
+        const payment = await db.payments.getPaymentByOrderNumber(ordernumber);
+        if (!payment) {
+            return res
+                .status(422)
+                .json('Payment details not available in the system.');
+        }
+
+        const dbUser = await db.users.getUser(user.phoneNumber);
+        let paymentDetails = {};
+        paymentDetails.amount = payment.amount;
+        paymentDetails.status = payment.payment_status;
+        paymentDetails.balance = dbUser.balance;
+        res.status(200).json(paymentDetails);
+    } catch (err) {
+        res
+            .status(500)
+            .json(`Failed to get payment details. Error: ${err.message}`);
+    }
+};
+
 const paymentNotify = async (req, res) => {
```

Comment:
We do have a paymentNotify request being sent by bambora. And based on the communication with Bambora they said that is not necessary. This can be used for our own system logging or debugging.

Reply:
If Bambora is not using it, I guess it could then be deleted as I doubt our frontend is either using it now?

Reply:
I recommend this can be used for server side logging. just in case. I don't feel like removing it :). I will remove if you strongly recommend.

Reply:
But if nothing is using it on the server at the moment, I don't think it should be left here. Logging otherwise is of course good so the log statement could then be moved somewhere? :)

Reply:
we are passing this notify_url to the bambora payment request. this is kind of mandatory in the request structure. So the bambora calls back on the notify url to the backend anyway, So I think it can be used :) Do you agree?.

Reply:
yes bambora calls two urls.
1. return_url (we are only using this as of now).
2. notify_url